client.go

```diff
@@ -1188,3 +1188,76 @@ func validateQueueName(queueName string) error {
 	}
 	return nil
 }
+
+// JobList returns a paginated list of jobs matching the provided filters. The
+// provided context is used for the underlying Postgres query and can be used to
+// cancel the operation or apply a timeout.
+//
+//	params := river.JobListParams{}.WithLimit(10).State(river.JobStateCompleted)
+//	jobRows, err := client.JobList(ctx, params)
+//	if err != nil {
+//		// handle error
+//	}
+func (c *Client[TTx]) JobList(ctx context.Context, params *JobListParams) ([]*rivertype.JobRow, error) {
+	if params == nil {
+		params = NewJobListParams()
+	}
+	// TODO(bgentry): confirm with Brandur, do we want to error in this scenario?
+	// If so this error is not worded appropriately for these methods.
+	if c.driver.GetDBPool() == nil {
+		return nil, errInsertNoDriverDBPool
+	}
```

Comment:
@brandur left this here to ask you. We will need to error for pseudo-drivers like the current `database/sql` one, right? Although this error seems specifically worded for insert. Should we make it more generalized?

Reply:
I tweaked the name and text to be a little more generalized, lmk your thoughts.